src/WalkingModule/src/Module.cpp

```diff
@@ -945,74 +827,60 @@ bool WalkingModule::updateModule()
         yawRotation = yawRotation.inverse();
         modifiedInertial = yawRotation * m_inertial_R_worldFrame;
 
-        if(m_useQPIK)
+        if (m_useQPIK)
         {
             // integrate dq because velocity control mode seems not available
             yarp::sig::Vector bufferVelocity(m_robotControlHelper->getActuatedDoFs());
             yarp::sig::Vector bufferPosition(m_robotControlHelper->getActuatedDoFs());
 
-            if(!m_FKSolver->setInternalRobotState(m_qDesired, m_dqDesired))
+            if (!m_FKSolver->setInternalRobotState(m_qDesired, m_dqDesired))
             {
                 yError() << "[WalkingModule::updateModule] Unable to set the internal robot state.";
                 return false;
             }
 
-            if (!m_useBLFIK)
-            {
-                if (!solveQPIK(m_QPIKSolver,
-                               desiredCoMPosition,
-                               desiredCoMVelocity,
-                               yawRotation,
-                               m_dqDesired))
-                {
-                    yError() << "[WalkingModule::updateModule] Unable to solve the QP problem with "
-                                "osqp.";
-                    return false;
-                }
-            } else
+            yawRotation = yawRotation * m_trajectoryGenerator->getChestAdditionalRotation();
```

Comment:
So the fact that we did not have this before was a bug on the usage of the BLF IK on old iCub robots?

Reply:
Yes 😞